salt/grains/core.py

```diff
@@ -2536,6 +2540,10 @@ def _hw_data(osdata):
                         grains[key] = salt.utils.stringutils.to_unicode(ifile.read().strip(), errors='replace')
                         if key == 'uuid':
                             grains['uuid'] = grains['uuid'].lower()
+                except UnicodeDecodeError:
+                    # Some firmwares provide non-valid 'product_name'
+                    # files, ignore them
+                    pass
```

Comment:
Does it make sense to write a log entry instead of silently `pass`. We should also avoid spamming the log files with a lot of messages. The core grains are executed often.

Reply:
Might be good as debug, or trace...